lib/Dialect/TritonGPU/IR/Dialect.cpp

```diff
@@ -247,6 +247,23 @@ SmallVector<unsigned> getWarpOrder(Attribute layout) {
   return order;
 }
 
+SmallVector<unsigned> getMFMADotOrder(unsigned opIdx, unsigned rank) {
```

Comment:
@antiagainst @zhanglx13 This is probably true for other mma layouts as well (wmma, nvidia layouts), but I wasn't sure, so I added more conservative change that only changes this for dot with MFMA parent. Order of dot layout becomes more important once you want to do global load in it, because of the vectorization that relies on axisInfo utilities that assume things about order mentioned in the comment. Since this wasn't done before, and we always loaded from HBM in blocked layout, order of dot was probably not important so this wasn't a problem.

Reply:
FYI @zhanglx13 recently landed https://github.com/triton-lang/triton/pull/4767 which toches this part. Looking at your comment below:

> Therefore, it's important to differentiate between order when opIdx == 0 and opIdx == 1, as both have the highest contiguity in the K dimension.

That's only true if the input is in NT format? IIUC by default matrices are all interpreted as row-major in the conceptual form so I guess that's why you are seeing the mismatch and need to swap.. Looks we have some implicit assumptions that need to be fixed? Correct me if I'm wrong though.

Reply:
Could you change the function name to `getOrderForDotOperand`?

Reply:
> This is probably true for other mma layouts as well (wmma, nvidia layouts),

I think it's true for mmav2